src/hard.js

```diff
@@ -0,0 +1,90 @@
+function fizzBuzz(start, end) {
+  let fizzBuzz = [];
+
+  for (let i = start; i < end; i++) {
+    if (i % 15 === 0) {
+      fizzBuzz.push("fizzbuzz");
+    } else if (i % 3 === 0) {
+      fizzBuzz.push("fizz");
+    } else if (i % 5 === 0) {
+      fizzBuzz.push("buzz");
+    } else {
+      fizzBuzz.push(i);
+    }
+  }
+  return fizzBuzz;
+}
+
+function hasDuplicate(numbers) {
+  let numberTally = {};
+
+  for (let number = 0; number < numbers.length; number++) {
+    if (numbers[number] in numberTally) {
+      return true;
+    } else {
+      numberTally[numbers[number]] = 1;
+    }
+  }
+
+  return false;
+}
+
+function findPlurality(items) {
+  if (items.length === 0) {
+    return undefined;
+  }
+
+  let itemCounter = {};
+
+  for (let i = 0; i < items.length; i++) {
+    if (items[i] in itemCounter) {
+      itemCounter[items[i]]++;
+    } else {
+      itemCounter[items[i]] = 1;
+    }
```

Comment:
refactor: we can avoid using else by switching this into

```suggestion
    if (!(items[i] in itemCounter)) {
      itemCounter[items[i]] = 0;
    }
    ++itemCounter[items[i]];
```

Reply:
can't believe you're just allowed to put a ! in front of anything

Reply:
I'll refactor all the times this logic shows up through medium.js and easy.js as well